### DISCUSSION TITLE
Automatic Socket Reconnect

### DISCUSSION THREAD

**Discussion by docelmo** [jchristn/WatsonWebsocket#85] [General]
Maybe this was addressed and maybe not but I figured I would ask after looking at the other topics.  Will this library automatically reconnect or do I need to send keep alives to keep the connection in place?

**Comment by jchristn** [comment]
Hi @docelmo the library will not automatically reconnect for you, however you can subscribe to ```ServerDisconnected``` events in case you want to re-connect.  I would generally advise having a background task that periodically sends a heartbeat message and validates connectivity if you want a long-lived connection.
